### PR TITLE
schematelemetry: don't redact object ID or validation error in logs

### LINKED ISSUE #126584: schematelemetry: better redaction for invalid objects
We log found invalid objects here
https://github.com/cockroachdb/cockroach/blob/c64b755ece7914767823aa18de42513716935fc6/pkg/sql/catalog/schematelemetry/schema_telemetry_job.go#L141-L151

In our live-streamed [logs](https://us5.datadoghq.com/logs?query=found%20invalid%20object%20with%20ID%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&event=AgAAAZBRnK0dlqQsvgAAAAAAAAAYAAAAAEFaQlJuSzBkQUFCM2R4a2o2bTRpNkFBRAAAACQAAAAAMDE5MDUxOWUtMmY2Ny00OTYxLThhYjktNzkyMTkzN2NmMTc5&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=flex_tier&stream_sort=time%2Casc&viz=stream&from_ts=1719350201431&to_ts=1719955001431&live=true), most of the information we need is redacted; so, we end up having to grab a debug zip

There might a better way to redact the validationErr so we do not have to do the dance above

Jira issue: CRDB-39987

Epic CRDB-39822

### PR DESCRIPTION
**Pull request by rafiss**
fixes: #126584
Release note: None

### CODE REVIEW

File: pkg/sql/catalog/schematelemetry/schema_telemetry_job.go [hunk 1]
```diff
@@ -148,7 +149,11 @@ func processInvalidObjects(
 				return errors.AssertionFailedf("expected err to be string (was %T)", row[1])
 			}
 
-			log.Warningf(ctx, "found invalid object with ID %d: %q", descID, validationErr)
+			// IDs are always non-sensitive, and the validationErr is written to the
+			// table with redact.Sprint, so it's also non-sensitive.
+			log.Warningf(ctx, "found invalid object with ID %d: %s",
```
Comment (annrpom) [inline pkg/sql/catalog/schematelemetry/schema_telemetry_job.go:154]: lgtm! do we have any testing for this?

**Review by annrpom — APPROVED** [review]

**Comment by rafiss** [comment]
tftr!

bors r+

**Pull request merged**